src/controlplane/scraper.go

```diff
@@ -0,0 +1,230 @@
+type Providers struct {
+	K8s kubernetes.Interface
+}
+
+// Scraper takes care of getting metrics from an autodiscovered CP instances.
+type Scraper struct {
+	Providers
+	logger               log.Logger
+	config               *config.Mock
+	k8sVersion           *version.Info
+	components           []Component
+	informerClosers      []chan<- struct{}
+	PodListerByNamespace map[string]v1.PodLister
+}
+
+// ScraperOpt are options that can be used to configure the Scraper
+type ScraperOpt func(s *Scraper) error
+
+// WithLogger returns an OptionFunc to change the logger from the default noop logger.
+func WithLogger(logger log.Logger) ScraperOpt {
+	return func(s *Scraper) error {
+		s.logger = logger
+		return nil
+	}
+}
+
+// Close will signal internal informers to stop running.
+func (s *Scraper) Close() {
+	for _, ch := range s.informerClosers {
+		close(ch)
+	}
+}
+
+// NewScraper builds a new Scraper, initializing its internal informers. After use, informers should be closed by calling
+func NewScraper(config *config.Mock, providers Providers, options ...ScraperOpt) (*Scraper, error) {
+	var err error
+	s := &Scraper{
+		config:    config,
+		Providers: providers,
+		// TODO: An empty implementation of the logger interface would be better
+		logger:               log.New(false, io.Discard),
+		PodListerByNamespace: make(map[string]v1.PodLister),
+	}
+
+	for i, opt := range options {
+		if err := opt(s); err != nil {
+			return nil, fmt.Errorf("applying config option #%d: %w", i, err)
+		}
+	}
+
+	// TODO If this could change without a restart of the pod we should run it each time we scrape data,
+	// possibly with a reasonable cache Es: NewCachedDiscoveryClientForConfig
+	s.k8sVersion, err = providers.K8s.Discovery().ServerVersion()
+	if err != nil {
+		return nil, fmt.Errorf("fetching K8s version: %w", err)
+	}
+
+	s.buildDiscoverer(s.buildComponents())
+
+	return s, nil
+}
+
+// Run scraper collect the data populating the integration entities
+func (s *Scraper) Run(i *integration.Integration) error {
+	for _, component := range s.components {
+
+		pod, err := s.discoverPod(component)
+		if err != nil {
+			return fmt.Errorf("control plane component %s discovery failed: %v", component.Name, err)
+		}
+
+		if pod == nil {
+			s.logger.Debugf("No pod found for component: %s", component.Name)
+			continue
+		}
+
+		grouper := grouper.New(
+			s.deprecatedClient(component, pod.Name),
+			component.Queries,
+			s.logger,
+			pod.Name,
+		)
+
+		job := scrape.NewScrapeJob(string(component.Name), grouper, component.Specs)
+
+		s.logger.Debugf("Running job: %s", job.Name)
+
+		result := job.Populate(i, s.config.ClusterName, s.logger, s.k8sVersion)
+
+		if len(result.Errors) > 0 {
+			s.logger.Infof("Error populating data from %s: %v", job.Name, result.Error())
+		}
+	}
+
+	return nil
+}
+
+func (s *Scraper) deprecatedClient(c Component, podName string) client.HTTPClient {
+	timeout := 500 * time.Millisecond
+
+	authMethod := controlplaneClient.None
```

Comment:
controlplane or controlPlane. The same comment apply to the whole file :)

Reply:
not sure about this convention, i though it was sub+package, 
`"github.com/newrelic/nri-kubernetes/v2/src/controlplane/client"` = `controlplaneClient`